hal/lib/athena/PWM.cpp

```diff
@@ -0,0 +1,172 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) FIRST 2016. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "HAL/PWM.h"
+
+#include "DigitalInternal.h"
+
+static_assert(sizeof(uint32_t) <= sizeof(void*),
+              "This file shoves uint32_ts into pointers.");
+
+namespace hal {
+// Create a mutex to protect changes to the DO PWM config
+priority_recursive_mutex digitalPwmMutex;
```

Comment:
What's the approach here?  Is this declared in DigitalInternal.h?  If so, why is it not in DigitalInternal.cpp?  If not, why is it not static?